solutions/usaco-574.mdx

```diff
@@ -52,60 +52,95 @@ int main() {
 Instead of a knapsack solution, a BFS/DFS can be done to iterate though **all** attainable states and simply choose the one with maximum fullness. The scene can be thought of as a graph by defining each edge to be the transition from fullness $i$ to $j$. Each state is defined by fullness and whether Bessie has drank water. Since there are $2T$ states and each state is visited at most once, the time and space complexity of this solution is $\mathcal{O}(T)$. Below is a BFS implementation.
 
 <LanguageSection>
-
 <CPPSection>
 
 ```cpp
-#include <bits/stdc++.h>
-using namespace std;
-
-const int maxn = 5e6 + 5;
-bool pass[maxn][2];
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <set>
+#include <queue>
 
-struct node {
-	int val;
-	bool drunk;
-};
+using std::cout;
+using std::endl;
+using std::vector;
+using std::pair;
 
-int bfs(int t, int a, int b) {
-	int res = INT_MIN;
-	queue<node> q;
-	q.push({0, 0});
-
-	while (!q.empty()) {
-		int value = q.front().val;
-		bool drunk = q.front().drunk;
-		q.pop();
-		res = max(res, value);
-
-		if(value + a <= t && !pass[value + a][drunk]) {
-			q.push({value + a, drunk});
-			pass[value + a][drunk] = true;
-		}
-
-		if(value + b <= t && !pass[value + b][drunk]) {
-			q.push({value + b, drunk});
-			pass[value + b][drunk] = true;
+int main() {
+	std::ifstream read("feast.in");
+	int max_fullness;
+	int orange;
+	int lemon;
+	read >> max_fullness >> orange >> lemon;
+
+	std::set<pair<int, bool>> poss_fullness{{0, false}};
+	std::queue<pair<int, bool>> queue;
```

Comment:
maybe don't name your data structure queue "queue". It can cause issues.

Reply:
that's why we have `std::`

Reply:
not everyone uses std::

Reply:
well, this solution does

Reply:
if they try to copy what you did into their code and they dont use std::, then they will get error and be sadddddd

Reply:
it's called reading error messages

Reply:
you can avoid them having to read the message in the first place

Reply:
> if they try to copy what you did into their code and they dont use std::, then they will get error and be sadddddd

also, it doesn't even error
```cpp
#include <iostream>
#include <queue>

using namespace std;

int main() {
	queue<int> queue;
	queue.push(0);
	cout << queue.front() << endl;
}
```
compiles & runs just fine

Reply:
![image](https://user-images.githubusercontent.com/79723780/169627912-d9254cc8-0de9-4d2c-b925-bf940a07bccd.png)

i dont make the rules

Reply:
you got me there